types/mkdirp/index.d.ts

```diff
@@ -1,15 +1,15 @@
-// Type definitions for mkdirp 0.5
-// Project: https://github.com/substack/node-mkdirp
+// Type definitions for mkdirp 1.0
+// Project: https://github.com/isaacs/node-mkdirp#readme
 // Definitions by: Bart van der Schoor <https://github.com/Bartvds>
 //                 mrmlnc <https://github.com/mrmlnc>
+//                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
 import fs = require('fs');
 
-declare function mkdirp(dir: string, cb: (err: NodeJS.ErrnoException, made: mkdirp.Made) => void): void;
-declare function mkdirp(dir: string, opts: mkdirp.Mode | mkdirp.Options, cb: (err: NodeJS.ErrnoException, made: mkdirp.Made) => void): void;
+declare function mkdirp(path: string, opts?: mkdirp.Mode): Promise<mkdirp.Made>;
```

Comment:
[The docs](https://github.com/isaacs/node-mkdirp#methods) suggest that `mkdirp` returns `string | undefined`. Why are we returning `Made` here?

Reply:
I'm after this change, would love to see it merged.

isn't the only problem that Made (line 15) should be `string | undefined` now?

and the code does return `string | undefined`: https://github.com/isaacs/node-mkdirp/blob/master/lib/find-made.js#L3

Reply:
I'll try to review my own code changes, I admit it was done quickly. Can you folks post me a snips of usage of updated version (v1) (via gists), thanks!

Reply:
I'm afraid my use of mkdirp 1.0 is the simplest imaginable:

```ts
await mkdirp("some dir");
```

Reply:
updated/amended